### PR TITLE
[SYCL] Fix long long support in vec::convert on Windows

### PR DESCRIPTION
**Pull request by dnmokhov**
- Keep `long long` and `unsigned long long` in `is_standard_type`.
  Excluding them resulted in duplicate `convertImpl` definitions on
  Windows when converting from `long long` or `unsigned long long`:
  SPIR-V-based and the non-standard type version.
- For to-int SPIR-V-based `convertImpl` overload resolution, compare
  the OpenCL return type with cl_DestType instead of simply DestType
  ((u)char, (u)short, (u)int, (u)long). When converting to `long long`
  or `unsigned long long` on Windows, the non-standard type version
  was used instead of these functions, because `cl_long == long long
  != long` there.

### CODE REVIEW

**Comment by romanovvlad** [comment]
@bader  @turinevgeny  Could you please review?

**Review by turinevgeny — APPROVED** [review]
LGTM,

File: sycl/test/basic_tests/vectors/vectors.cpp [hunk 1]
```diff
@@ -82,5 +109,14 @@ int main() {
   assert((std::is_same<cl::sycl::vec<unsigned long, 8>, cl::sycl::ulong8>::value));
   assert((std::is_same<cl::sycl::vec<unsigned long, 16>, cl::sycl::ulong16>::value));
 
+  // Check convert() from and to various types.
+  check_convert_from<int8_t>();
+  check_convert_from<int16_t>();
+  check_convert_from<int32_t>();
+  check_convert_from<int64_t>();
```
Comment (romanovvlad) [inline sycl/test/basic_tests/vectors/vectors.cpp:116]: Should we have line which `long long` here?

File: sycl/test/basic_tests/vectors/vectors.cpp [hunk 2]
```diff
@@ -10,21 +10,48 @@
 
 #define SYCL_SIMPLE_SWIZZLES
 #include <CL/sycl.hpp>
-using namespace cl::sycl;
 
-void check_vectors(int4 a, int4 b, int4 c, int4 gold) {
-  int4 result = a * (int4)b.y() + c;
+void check_vectors(cl::sycl::int4 a, cl::sycl::int4 b, cl::sycl::int4 c,
```
Comment (bader) [inline sycl/test/basic_tests/vectors/vectors.cpp:14]: I suggest dropping `cl::` prefix for all instances of `cl::sycl::` namespace qualifier.

E.g. `cl::sycl::int4` -> `sycl::int4`.

File: sycl/test/basic_tests/vectors/vectors.cpp (hunk 1 above)
Comment (dnmokhov) [reply sycl/test/basic_tests/vectors/vectors.cpp:116]: On Windows, `std::is_same_v<int64_t,  long long>`, so the test covers the fixed issue.

I guess I can still add `char`, `short`, `int`, `long`, `long long` for more coverage though.

**Review by romanovvlad — APPROVED** [review]

**Review by bader — APPROVED** [review]

**Pull request merged**